### ISSUE TITLE
Push new release through official channels for vWii users

### ISSUE THREAD

**Issue by fincs** [fail0verflow/hbc#10]
I just set up HBC on my Wii U's vWii using the HackMii Installer. As it is well known, HBC has an aspect ratio bug when running on vWii; and the open source release [fixed said bug as a "bonus"](https://hackmii.com/2016/11/the-open-homebrew-channel/). However, currently it is not possible to install this fixed version of HBC through official means (neither the HackMii Installer nor the server HBC automatically checks for updates have been updated with a fixed HBC build). So currently it is necessary to rely on third party tools and websites (some of which are heavily associated with piracy) to install either an unofficially built HBC (as a .wad, yuck!) or a forwarder channel (even more yuck!).

Therefore, I'm asking if it would be possible to push a new release of HBC through official channels; and make it possible to avoid relying on dodgy third parties to get fully functional vWii homebrew running on a Wii U.

**Comment by HTV04** [comment]
Don't know if you've tried this yet, but here's an aspect fixer app from Tueidj:

http://www.tueidj.net/aspect_fix.zip

He worked on Riivolution, a popular patcher for Wii games.

**Comment by HTV04** [comment]
Hey, quick update: I found the problem, it's the title ID.

The L in "LULZ" throws the vWii off and makes it think that the HBC is a Sega Master System Channel. It then applies letterboxing.

By updating the main HBC _just_ to change the title ID to that of the OHBC (which, fittingly, has a title ID of "OHBC"), this problem could be solved. A lot of homebrew already support the new title ID, so it could work.

However, if you have concerns that this might clash with people who've already installed the OHBC, then please release a compiled (vWii compatible?) WAD.

**Comment by fancythedeveloper** [comment]
Just use the compiled binaries of @FIX94's fork.

**Comment by fincs** [comment]
I'm not interested in third party forks, or having to deal with .wad installers.

**Comment by HTV04** [comment]
FIX94 is an extremely reputable person in the vWii community, though. He's developed loads of things that have increased compatibility with vWii. I'm going to have to agree with @fancythedeveloper for the time being.

A good WAD installer is Wii Mod Lite, by the way. I have encountered almost no issues with it, and it's extremely easy to use.

**Comment by DacoTaco** [comment]
thats not the point. a user shouldn't use 3th party tools or fork's if the official one is around.
imo fincs is right and this would need a new updated release.

**Comment by fincs** [comment]
Also, I have no desire or intention of using WAD managers, or similar piracy related tools.

**Comment by HTV04** [comment]
@DacoTaco I agree, there should be an official update. A forced update may cause problems with some homebrew with the new ID, so they should either make the update optional, or add a hidden forwarder channel with the old ID.

**Comment by DacoTaco** [comment]
i think youre missing a few things.
i don't think HBC updates have even once been forced.
And no, HBC should not be backwards compatible in its titleID. all homebrew run from HBC don't need to be aware of what HBC version is installed. all they have to do is exit (libogc will check on exit if the HBC reload stub is there and will run HBC through that if it is).

Only Priiloader should be made aware of what to look out for with HBC versions & titles, but all homebrew loaded from that shouldn't ( cause just like HBC, it sets up stuff for libogc)

so no, the update should be pushed like they always have (optional?), should overwrite or replace the current one with a different ID and shouldn't use a hidden forwarder channel ( eeuw, forwarder channel. nasty )

**Comment by marcan** [comment]
The title ID is not a problem. We would not change the title ID unless Nintendo banned it. The aspect ratio thing is already fixed [here](https://github.com/fail0verflow/hbc/blob/master/channel/channelapp/source/gfx.c#L58). The fix has nothing to do with the title ID. The title ID triggers a different default video configuration, but nothing says you can't change it back and keep it.

The reason why we aren't pushing an official update is because it would just annoy the vast majority of users (non-vwii) who don't care, and would expose them to potential regressions for no benefit. The reason why we aren't *building* and not pushing an official update (i.e. release via manual download only) is because this is not just HBC itself, but the whole HackMii Installer, including BootMii, and *that* code is way more delicate. Every time we update HBC we've had to deal with toolchain/library regressions, and I ran into some stuff with the open source release too. I don't have the bandwidth to do a full update of everything and regression test it to our standards right now.

If we had a hermetic build environment it would be easier, but sadly, although our internal repo is pretty nice and the build environment makes it easy to build all the pieces and make a release, all the devkitppc/libogc/3rdparty stuff is out-of-band (plus a separate toolchain for BootMii bits and pieces is also involved). I probably should've figured out some way to version that to make golden reproducible builds possible and thus make it less scary to fix just the vwii thing for people, but I didn't, and now it's been several years and I'm not looking forward to running through testing everything again, and I refuse to make an official release not up to testing standards. Sorry.
